isamples_metadata/OpenContextTransformer.py

```diff
@@ -0,0 +1,217 @@
+            cls._rockMapper,
+        ]
+
+
+class SpecimenCategoryMetaMapper(AbstractCategoryMetaMapper):
+    _organismPartMapper = StringEqualityCategoryMapper(
+        [
+            "Animal Bone",
+            "Human Bone",
+            "Non Diagnostic Bone",
+        ],
+        "Organism part",
+    )
+    _anthropogenicAggregationMapper = StringEqualityCategoryMapper(
+        ["Architectural Element", "Basket", "Bulk Ceramic", "Lot"],
+        "Anthropogenic aggregation",
+    )
+    _biomeAggregationMapper = StringEqualityCategoryMapper(
+        ["Biological subject, Ecofact", "Plant remains"], "Biome aggregation"
+    )
+    _artifactMapper = StringEqualityCategoryMapper(
+        ["Coin", "Glass", "Groundstone", "Object", "Pottery", "Sculpture", "Stela"],
+        "Artifact",
+    )
+    _otherSolidObjectMapper = StringEqualityCategoryMapper(
+        ["Sample", "Sample, Collection, or Aggregation"], "Other solid object"
+    )
+    _organismProductMapper = StringEqualityCategoryMapper(
+        [
+            "Shell",
+        ],
+        "Organism product",
+    )
+
+    @classmethod
+    def categories_mappers(cls) -> typing.List[AbstractCategoryMapper]:
+        return [
+            cls._organismPartMapper,
+            cls._anthropogenicAggregationMapper,
+            cls._biomeAggregationMapper,
+            cls._artifactMapper,
+            cls._otherSolidObjectMapper,
+            cls._organismProductMapper,
+        ]
+
+
+class OpenContextTransformer(Transformer):
+    N2T_PREFIX = "https://n2t.net/"
+    N2T_ARK_PREFIX = f"{N2T_PREFIX}ark:/"
+
+    def transform(self) -> typing.Dict:
+        transformed_record = super().transform()
+        return transformed_record
+
+    def _citation_uri(self) -> typing.AnyStr:
+        return self.source_record.get("citation uri")
+
+    def id_string(self) -> typing.AnyStr:
+        citation_uri = self._citation_uri()
+        return f"metadata/{citation_uri.removeprefix(self.N2T_ARK_PREFIX)}"
+
+    def sample_identifier_scheme(self) -> typing.AnyStr:
+        pass
+
+    def sample_identifier_string(self) -> typing.AnyStr:
+        return self._citation_uri().removeprefix(self.N2T_PREFIX)
+
+    def sample_identifier_value(self) -> typing.AnyStr:
+        # TODO is this even used?
+        return self._citation_uri().removeprefix(self.N2T_PREFIX)
+
+    def sample_label(self) -> typing.AnyStr:
+        return self.source_record.get("label", Transformer.NOT_PROVIDED)
+
+    def sample_description(self) -> typing.AnyStr:
+        description_pieces = []
+        self._transform_key_to_label(
+            "early bce/ce", self.source_record, description_pieces
+        )
+        self._transform_key_to_label(
+            "late bce/ce", self.source_record, description_pieces
+        )
+        self._transform_key_to_label("updated", self.source_record, description_pieces)
+        self._transform_key_to_label("updated", self.source_record, description_pieces)
```

Comment:
Repeated yes?

Reply:
Oh yes, and I saw this in the test output too.  Thanks!